commodore/cluster.py

```diff
@@ -13,14 +13,79 @@
 )
 
 from .config import Config
-
-
-def fetch_cluster(cfg, clusterid):
-    cluster = lieutenant_query(cfg.api_url, cfg.api_token, "clusters", clusterid)
-    # TODO: move Commodore global defaults repo name into Lieutenant
-    # API/cluster facts
-    cluster["base_config"] = "commodore-defaults"
-    return cluster
+from .git import HEAD
+
+
+class Cluster:
+    def __init__(
+        self,
+        cfg: Config,
+        cluster_response: Dict,
+        tenant_response: Dict,
+    ):
+        self.__cfg = cfg
+        self.__cluster = cluster_response
+        self.__tenant = tenant_response
+        if (
+            "tenant" not in self.__cluster
+            or self.__cluster["tenant"] != self.__tenant["id"]
+        ):
+            raise click.ClickException("Customer id mismatch")
+
+    def id(self) -> str:
```

Comment:
```suggestion
    @property
    def id(self) -> str:
```

By decorating getters with `@property` they can be used as `cluster.id` instead of `cluster.id()`.

Reply:
But then they can also be changed by assigning values right? Maybe I am to conservative here but I like this to not be possible.

Reply:
You can't assign to a method with `@property`, you'd need to implement (taking `id` as the example):

```
@id.setter
def set_id(self, id):
    # do whatever is necessary to update the id
    pass
```

See https://docs.python.org/3/library/functions.html#property

Reply:
That then raises the question: should all functions of `Cluster` be properties? My gut feelings tell me, this is true for `id`, `tenant` and `facts` but not for the others. But I can not explain why.

Reply:
From what I can tell, the whole `Cluster` class is a construct which exposes "properties" of the cluster which we're compiling a catalog for, therefore I'd argue that all the current function (except the constructor obviously) could be made properties.